scripts/creation-portal-migration/orgDetails.js

```diff
@@ -0,0 +1,83 @@
+require("dotenv").config({ path: "./../../.env" });
+const { createDBInstance } = require("./db/dbConfig");
+const { findAll } = require("./db");
+const { CONFIG } = require("./constant/config");
+const { compact, uniq, uniqBy } = require("lodash");
+const fs = require("fs");
+const { searchUser, getOpenSaberUserOrgId } = require("./api-list/user");
+
+const getUserIds = async () => {
+  try {
+    const db = await createDBInstance();
+    const data = await findAll(CONFIG.DB.TABLES.solutions, {
+      programId: { $exists: true },
+      type: { $in: ["observation", "survey"] },
+    });
+
+    const userIds = data.map((solution) => solution.author);
+    const solutions = data.map((solution) => {
+      // const months = ["JAN", "FEB", "MAR", "APR", "MAY", "JUNE", "JULY", "AUG", "SEP", "OCT", "NOV", "DEC"];
+      // const month = months[new Date().getMonth()];
+      // const day = new Date().getDate();
+      // const year = new Date().getFullYear();
+      const programName = `${solution.name} sourcing project`
+      return {
+        userId: solution.author,
+        solutionId: solution._id,
+        solutionName: solution.name,
+        programId: solution?.migrationReference?.sourcingProgramId || "",
+        programName: programName,
+      };
+    });
+    let uniqUsers = uniq(userIds);
+    uniqUsers = compact(uniqUsers);
+    const usersList = await searchUser(uniqUsers);
+    const openSaberOrg = await getOpenSaberUserOrgId();
+    const d = writeToCSVFile(usersList, uniqUsers, openSaberOrg, solutions);
+    console.log(`\n migratedCount userIds`, d);
+  } catch (err) {
+    console.log(`Error while migrating : ${err}`);
+    throw new Error("Error occured", err);
+  }
+};
+
+writeToCSVFile = (users, usersIdsInDb, openSaberOrg, solutions) => {
+  const filename = __dirname + "/SL-DataMapping.csv";
+  fs.writeFile(
+    filename,
+    extractAsCSV(users, usersIdsInDb, openSaberOrg, solutions),
+    (err) => {
+      if (err) {
+        console.log("Error writing to csv file", err);
+      } else {
+        console.log(`saved as ${filename}`);
+        process.exit();
+      }
+    }
+  );
+};
+
+const extractAsCSV = (users, usersIdsInDb, openSaberOrg, solutions) => {
+  const header = [
+    `authorId,mappedUserId,userName,rootOrgId,rootOrgName,org_id,srcOrgAdminId,srcOrgAdminUserName,contributorOrgAdminId,contributorOrgAdminUserName,programId,programName`,
```

Comment:
@swetha-aggidevara from where we will be getting srcOrgAdminId, srcOrgAdminUserName & contributorOrgAdminId ?

Reply:
@VISHNUDAS-tunerlabs  As we discussed in the call, we have to an API to update it manually, Please check with piyush for more information.